pilot/pkg/model/config.go

```diff
@@ -621,18 +621,12 @@ func (store *istioConfigStore) routeRulesByDestinationV2(instances []*ServiceIns
 	return nil
 }
 
-func (store *istioConfigStore) EgressRules() map[string]*routing.EgressRule {
-	out := make(map[string]*routing.EgressRule)
-	rs, err := store.List(EgressRule.Type, NamespaceAll)
+func (store *istioConfigStore) EgressRules() []Config {
+	configs, err := store.List(EgressRule.Type, NamespaceAll)
 	if err != nil {
 		return nil
 	}
-	for _, r := range rs {
-		if rule, ok := r.Spec.(*routing.EgressRule); ok {
-			out[r.Key()] = rule
-		}
-	}
-	return out
+	return configs
 }
 
 func (store *istioConfigStore) ExternalServices() []Config {
```

Comment:
High level comment: please focus on external services. Egress rules are going away soon. External service code is active

Reply:
more like undo changes to egressrules code. Its going to be hard to track both..

Reply:
It is much easier to test first with egress rule, which works and is tested. Let's get this in and I will remove everything when egress rule is removed anyway.